src/js/common/components/DropDown.jsx

```diff
@@ -6,6 +6,26 @@ class DropDown extends React.Component {
   constructor(props) {
     super(props);
     this.toggleDropDown = this.toggleDropDown.bind(this);
+    this.handleDocumentClick = this.handleDocumentClick.bind(this);
+  }
+
+  componentWillMount() {
+    this.props.container.addEventListener('click', this.handleDocumentClick, false);
+    this.props.container.addEventListener('touchend', this.handleDocumentClick, false);
+  }
+
+  componentWillUnmount() {
+    this.props.container.removeEventListener('click', this.handleDocumentClick, false);
+    this.props.container.removeEventListener('touchend', this.handleDocumentClick, false);
+  }
+
+  handleDocumentClick(event) {
+    // If this dropdown is open, and it's not an element within this dropdown being clicked,
+    // then the user clicked elsewhere and we should invoke the click handler to toggle this
+    // dropdown to closed.
+    if (this.props.isOpen && !this.dropdownDiv.contains(event.target)) {
+      this.props.clickHandler();
```

Comment:
Perhaps `this.toggleDropdown()` for clarity. They may be the same right now, but `clickHandler` isn't used directly anywhere else, and it would be cleaner to bind both this event and the button click to the same function.

Reply:
That makes sense for sure. Will update that!